Userland/Libraries/LibGfx/PNGLoader.cpp

```diff
@@ -871,7 +871,9 @@ static ErrorOr<void> process_iCCP(ReadonlyBytes data, PNGLoadingContext& context
     if (compression_method != 0)
         return Error::from_string_literal("Unknown profile compression method");
 
-    context.embedded_icc_profile = EmbeddedICCProfile { { data.data(), profile_name_length }, data.slice(profile_name_length + 2) };
+    auto profile_name = TRY(ByteBuffer::copy(data.trim(profile_name_length)));
+    auto compressed_data = TRY(ByteBuffer::copy(data.slice(profile_name_length + 2)));
+    context.embedded_icc_profile = EmbeddedICCProfile { profile_name, compressed_data };
```

Comment:
It's not clear why this is needed, please add context to the commit message. Also `move()` these into `EmbeddedICCProfile`, `ByteBuffer` still has a copy ctor sadly.